control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -2051,6 +2071,88 @@ func (r *reconciler) ensureGuestAdmissionWebhooksAreValid(ctx context.Context) e
 	return errors.NewAggregate(errs)
 }
 
+func (r *reconciler) reconcileKubeletConfig(ctx context.Context) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	wantCMList := &corev1.ConfigMapList{}
+	opts := &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{nodepool.KubeletConfigConfigMapLabel: "true"}),
+		Namespace:     r.hcpNamespace,
+	}
+	if err := r.cpClient.List(ctx, wantCMList, opts); err != nil {
+		return fmt.Errorf("failed to list KubeletConfig ConfigMaps from controlplane namespace %s: %w", r.hcpNamespace, err)
+	}
+
+	haveCMList := &corev1.ConfigMapList{}
+	opts = &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{nodepool.KubeletConfigConfigMapLabel: "true"}),
+		Namespace:     ConfigManagedNamespace,
+	}
+	if err := r.client.List(ctx, haveCMList, opts); err != nil {
+		return fmt.Errorf("failed to list KubeletConfig ConfigMaps from hostedcluster namespace %s: %w", ConfigManagedNamespace, err)
+	}
+
+	want := set.Set[string]{}
+	for _, cm := range wantCMList.Items {
+		want.Insert(cm.Name)
+	}
+
+	have := set.Set[string]{}
+	for _, cm := range haveCMList.Items {
+		have.Insert(cm.Name)
+	}
+
+	toCreate := want.Difference(have)
+	if len(toCreate) > 0 {
+		log.Info("create mirrored configs", "toCreate", toCreate.UnsortedList())
+	}
+	toDelete := have.Difference(want)
+	if len(toDelete) > 0 {
+		log.Info("delete mirrored configs", "toDelete", toDelete.UnsortedList())
+	}
+
+	for i := range haveCMList.Items {
+		cm := &haveCMList.Items[i]
+		if toDelete.Has(cm.Name) {
+			err := r.client.Delete(ctx, cm)
+			if err != nil && !apierrors.IsNotFound(err) {
```

Comment:
why do we need to check is notfound error here? delete is an idempotent operation

Reply:
any reason to not use DeleteIfNeeded from supportutil?